### PR TITLE
HSEARCH-2648 Add debug/trace logs showing execution time of Elasticsearch requests

### PR DESCRIPTION
**Pull request by yrodiere**
https://hibernate.atlassian.net/browse/HSEARCH-2648

### CODE REVIEW

**Comment by yrodiere** [comment]
@Sanne Rebased, it should work fine now :)

**Comment by Sanne** [comment]
Are you sure that making LuceneLogCategories extensible (rather than an enum) isn't a bit too much?

Was it because you didn't want to add Elasticsearch specific categories to the enum in hibernate-search-engine?

I see it as a nice move as other extension points - like Infinispan - will be able to create their own categories but I don't think an external integrator will actually do this, as they'll have their own logging conventions.

On the other hand, the previous `enum` approach had the nice effect of giving us (and users) a single place to look to have a list of all categories.

File: elasticsearch/src/main/java/org/hibernate/search/elasticsearch/client/impl/DefaultElasticsearchClient.java [hunk 1]
```diff
@@ -59,6 +67,10 @@ public Response execute(ElasticsearchRequest request) throws IOException {
 			 */
 			return e.getResponse();
 		}
+		finally {
+			long executionTime = System.currentTimeMillis() - start;
```
Comment (Sanne) [inline elasticsearch/src/main/java/org/hibernate/search/elasticsearch/client/impl/DefaultElasticsearchClient.java]: To measure durations don't use `System.currentTimeMillis()`, use System.nanoTime()

Comment (yrodiere) [inline elasticsearch/src/main/java/org/hibernate/search/elasticsearch/client/impl/DefaultElasticsearchClient.java]: Right; fixed.

**Comment by yrodiere** [comment]
> Are you sure that making LuceneLogCategories extensible (rather than an enum) isn't a bit too much?
> 
> Was it because you didn't want to add Elasticsearch specific categories to the enum in hibernate-search-engine?

Yes.

> I see it as a nice move as other extension points - like Infinispan - will be able to create their own categories but I don't think an external integrator will actually do this, as they'll have their own logging conventions.

I did not expect them to do this either, to be honest. Just wanted to avoid leaky abstractions internally.

> On the other hand, the previous enum approach had the nice effect of giving us (and users) a single place to look to have a list of all categories.

It doesn't seems _very_ hard to get a listing of all categories; Ctrl + Shift + G with my cursor on the constructor of `LogCategory` does the trick in Eclipse.

Do you want me to revert the change? I find this cleaner, but either way it will work.

**Comment by Sanne** [comment]
> Do you want me to revert the change? I find this cleaner, but either way it will work.

No, it's ok.

I just wanted to explore the reasoning with you. BTW I don't expect all users to always work with sources open, so we'll have to create a section in the documentation.

I'll merge this when back from lunch.. didn't go yet ;)

**Comment by yrodiere** [comment]
> so we'll have to create a section in the documentation

Pre-existing loggers are already documented [here](https://docs.jboss.org/hibernate/search//5.7/reference/en-US/html_single/#_optimizing_the_query_process) and [here](https://docs.jboss.org/hibernate/search//5.7/reference/en-US/html_single/#lucene-infostream). And I added a section in the Elasticsearch docs in this PR.
But maybe we should have a dedicated section, and make those I just mentioned redirect there?

**Comment by Sanne** [comment]
> But maybe we should have a dedicated section, and make those I just mentioned redirect there?

Yes that's what I meant. No rush though, just adding ideas for eventually reorganizing the docs.

**Comment by Sanne** [comment]
merged! thanks

**Pull request merged**